app/views/AnalyticalFramework/AssistedTagging/WidgetTagList/Matrix1dTagInput/index.tsx

```diff
@@ -0,0 +1,168 @@
+import React, { useMemo, useCallback } from 'react';
+import {
+    _cs,
+    isDefined,
+} from '@togglecorp/fujs';
+import {
+    ListView,
+} from '@the-deep/deep-ui';
+
+import {
+    Matrix1dWidget,
+    CategoricalMappingsItem,
+} from '#types/newAnalyticalFramework';
+import { sortByOrder } from '#utils/common';
+
+import CheckButton from '../../CheckButton';
+import CellGroup from '../../CellGroup';
+
+import styles from './styles.css';
+
+interface CellItem {
+    subRowKey: string;
+    subRowLabel: string;
+    rowOrder: number;
+    subRowOrder: number;
+    rowKey: string;
+    rowLabel: string;
+}
+
+const cellKeySelector = (cell: CellItem) => cell.subRowKey;
+const groupKeySelector = (n: CellItem) => n.rowKey;
+
+interface Props {
+    className?: string;
+    widget: Matrix1dWidget;
+    mappings: CategoricalMappingsItem[] | undefined;
+    onMappingsChange: (newMappings: CategoricalMappingsItem[], widgetPk: string) => void;
+    selectedTag: string | undefined;
+}
+
+function Matrix1dTagInput(props: Props) {
+    const {
+        className,
+        widget,
+        mappings,
+        onMappingsChange,
+        selectedTag,
+    } = props;
+
+    const sortedCells = useMemo(() => (
+        sortByOrder(widget?.properties?.rows)?.map((row) => {
+            const cells = row.cells.map((cell) => ({
+                rowKey: row.key,
+                rowOrder: row.order,
+                rowLabel: row.label,
+                subRowKey: cell.key,
+                subRowOrder: cell.order,
+                subRowLabel: cell.label,
+            }));
+            return cells;
+        }).flat()
+    ), [widget?.properties?.rows]);
+
+    const handleCellClick = useCallback((cellKey: string) => {
+        if (!selectedTag) {
+            return;
+        }
+
+        const selectedMappingsIndex = mappings?.findIndex((mapping) => {
```

Comment:
The mappings is filtered a lot of times. We need to see if we can avoid these repetitions.